### PR TITLE
#70 updated ConditionalWait methods signature.

### PR DESCRIPTION
**Pull request by knysh**
Added waitFor method which return true or false

### CODE REVIEW

File: src/test/java/aquality/selenium/waitings/ConditionalWaitTests.java [hunk 1]
```diff
@@ -155,26 +141,80 @@ public void testExceptionShouldBeCatchedConditionIsMetAndDefaultTimeoutIsNotOver
...
+        }, waitForTimeoutCondition, waitForTimeoutPolling, "Timeout exception should not be thrown");
         DurationSample durationSample = new DurationSample(timer.duration(), waitForTimeoutCondition);
-        BrowserManager.getBrowser().quit();
-
         assertTrue(durationSample.getDuration() < waitForTimeoutCondition);
+        assertTrue(conditionResult, "Condition result should be true");
+    }
+
+    @Test
+    public void testFalseShouldNotBeReturnedIfConditionIsNotMetAndTimeoutIsOver(){
```
Comment (DmitryBogatko) [inline src/test/java/aquality/selenium/waitings/ConditionalWaitTests.java]: please, rename method name as 'testfalseshould BE RETURNED'

File: src/test/java/aquality/selenium/waitings/ConditionalWaitTests.java [hunk 2]
```diff
@@ -155,26 +141,80 @@ public void testExceptionShouldBeCatchedConditionIsMetAndDefaultTimeoutIsNotOver
...
+            timer.start();
+            return false;
+        }, waitForTimeoutCondition, waitForTimeoutPolling, "Timeout exception should not be thrown");
+        DurationSample durationSample = new DurationSample(timer.duration(), waitForTimeoutCondition, defaultDeviation);
+        assertTrue(durationSample.isDurationBetweenLimits(), durationSample.toString());
+        assertFalse(conditionResult, "Condition result should be false");
+    }
+
+    @Test
+    public void testTrueShouldNotBeReturnedIfConditionIsMetAndDefaultTimeoutIsNotOver(){
```
Comment (DmitryBogatko) [inline src/test/java/aquality/selenium/waitings/ConditionalWaitTests.java]: rename as 'testTrueShouldBeReturned'

File: src/test/java/aquality/selenium/waitings/ConditionalWaitTests.java [hunk 3]
```diff
@@ -155,26 +141,80 @@ public void testExceptionShouldBeCatchedConditionIsMetAndDefaultTimeoutIsNotOver
...
+            timer.start();
+            return true;
+        }, "Timeout exception should not be thrown");
+        DurationSample durationSample = new DurationSample(timer.duration(), getTimeoutConfig().getCondition());
+        assertTrue(durationSample.getDuration() < getTimeoutConfig().getCondition());
+        assertTrue(conditionResult, "Condition result should be true");
+    }
+
+    @Test
+    public void testFalseShouldNotBeReturnedIfConditionIsNotMetAndDefaultTimeoutIsOver(){
```
Comment (DmitryBogatko) [inline src/test/java/aquality/selenium/waitings/ConditionalWaitTests.java]: rename as 'testFalseShouldBeReturned'

**Comment by DmitryBogatko** [comment]
increase additionally version in the pom.xml
add corrections into documentation files

File: src/test/java/aquality/selenium/waitings/ConditionalWaitTests.java (hunk 1 above)
Comment (knysh) [reply src/test/java/aquality/selenium/waitings/ConditionalWaitTests.java]: fixed

File: src/test/java/aquality/selenium/waitings/ConditionalWaitTests.java (hunk 2 above)
Comment (knysh) [reply src/test/java/aquality/selenium/waitings/ConditionalWaitTests.java]: fixed

File: src/test/java/aquality/selenium/waitings/ConditionalWaitTests.java (hunk 3 above)
Comment (knysh) [reply src/test/java/aquality/selenium/waitings/ConditionalWaitTests.java]: fixed

File: src/test/java/aquality/selenium/waitings/ConditionalWaitTests.java [hunk 4]
```diff
@@ -155,26 +141,80 @@ public void testExceptionShouldBeCatchedConditionIsMetAndDefaultTimeoutIsNotOver
...
+                    {
+                        timer.start();
+                        return true;
+                    }, waitForTimeoutCondition, waitForTimeoutPolling,
+                    "Conditional should be true", Collections.singleton(IllegalArgumentException.class));
+            DurationSample durationSample = new DurationSample(timer.duration(), waitForTimeoutCondition);
+            assertTrue(durationSample.getDuration() < waitForTimeoutCondition);
+            assertTrue(conditionResult, "Condition result should be true");
+        } finally {
+            BrowserManager.getBrowser().quit();
```
Comment (mialeska) [inline src/test/java/aquality/selenium/waitings/ConditionalWaitTests.java]: Maybe it's better to do this inside the cleanUp method

File: src/test/java/tests/usecases/FileDownloadingTests.java [hunk 5]
```diff
@@ -38,12 +36,6 @@ public void testDownloading() {
...
 
         BrowserManager.getBrowser().getDriver().switchTo().window(tabs.get(0));
-        ExpectedCondition<Boolean> fileDownloadedExpectedCondition = webDriver -> FileUtil.isFileDownloaded(fileAddress, lblFileContent);
-        try {
-            ConditionalWait.waitFor(fileDownloadedExpectedCondition, String.format("File %1$s should be downloaded", fileAddress));
-        } catch (TimeoutException e) {
-            BrowserManager.getBrowser().quit();
-            ConditionalWait.waitFor(fileDownloadedExpectedCondition, String.format("File %1$s should be downloaded", fileAddress));
-        }
+        Assert.assertTrue(ConditionalWait.waitFor(() -> FileUtil.isFileDownloaded(fileAddress, lblFileContent), String.format("File %1$s should be downloaded", fileAddress)));
```
Comment (mialeska) [inline src/test/java/tests/usecases/FileDownloadingTests.java]: is the retry currently works without the quit?

**Pull request merged**